HW_6/HW_7.py

```diff
@@ -0,0 +1,248 @@
+                for key, value in d.items():
+                    writer = csv.writer(words_count_file, delimiter='-')
+                    writer.writerow([f'{key}', f'{value}'])
+
+            with open('letters_count.csv', "w", newline='') as letters_count_file:
+                headers = ['Letter', 'Count_all', 'Count_uppercase', 'Percentage']
+                writer = csv.DictWriter(letters_count_file, fieldnames=headers)
+                writer.writeheader()
+                letters_lowercase = Counter(letter for line in content
+                                            for letter in line.lower()
+                                            if letter.isalpha())
+                letters_general = Counter(letter for line in content
+                                            for letter in line
+                                            if letter.isalpha())
+                letters_total = len(re.findall("[a-zA-Z]", content))
+                for letter, values in letters_lowercase.items():
+                    if letters_general.get(letter.upper()) is None:
+                        count_upper = 0
+                    else: count_upper = letters_general.get(letter.upper())
+                    writer.writerow({f'Letter': letter, f'Count_all': letters_lowercase.get(letter),
+                                     f'Count_uppercase': count_upper,
+                                     f'Percentage': round((letters_lowercase.get(letter) / letters_total * 100),
+                                                              2)})
+
+
+class News(Article):
+    def __init__(self, header, text, location):
+        Article.__init__(self, header=header, text=text)
+        self.location = location
+
+    def format_text(self, header, text, location):
+        return f"{header}--------------" \
+               f"\n{text}\n" \
+               f"{location}, {datetime.datetime.today().strftime('%d/%m/%Y %H.%M')}" \
+               f"\n{'-' * 23}\n\n"
+
+
+class PrivatAdv(Article):
+    def __init__(self, header, text, expiration_date):
+        Article.__init__(self, header=header, text=text)
+        self.expiration_date = expiration_date
+
+    def format_text(self, header, text, expiration_date):
+        return f"{header}----------" \
+               f"\n{text}\n" \
+               f"Actual until: {expiration_date}, {(expiration_date - datetime.date.today()).days} days left." \
+               f"\n{'-' * 23}\n\n"
+
+
+class Consider(Article):
+    def __init__(self, header, text, score):
+        Article.__init__(self, header=header, text=text)
+        self.score = score
+
+    def format_text(self, header, text, score):
+        return f"{header}-------------- " \
+               f"\n{text}\n" \
+               f"Probability of truth: {score}%" \
+               f"\n{'-' * 23}\n\n"
+
+
+def write_article():
+    header = Article.get_header()
+    text = Article.get_text()
+    if header == 'News':
+        if len(text) == 0:
```

Comment:
I'd add a message for the user that he/she should type Enter 2 times in order to go to JSON/XML parsing